api/models/v0.2/modelResponse.js

```diff
@@ -1,5 +1,21 @@
 const url = require('url');
 
+
+/**
+ * tryCatchParseInt(string)
+ * Model the response from creating a new patron.
+ *
+ * @param {string} string
+ * @return {number || string}
+ */
+function tryCatchParseInt(string) {
+  try {
+    return parseInt(string, 10);
+  } catch (error) {
+    return 'unparseable';
+  }
+}
+
```

Comment:
As written, you'll never hit line 16 above because `parseInt` does not ever throw anything:

```
> [undefined, null, '', '1', 'a', 3.14, true, false, [], {}, 1, new Object(), new Error()].map(parseInt)
[ NaN, NaN, NaN, 1, NaN, 3, NaN, NaN, NaN, NaN, 1, NaN, NaN ]
```

There's no reason to wrap `parseInt` in a try-catch. For that matter, there's no reason to run the regex test if you're just going to return 'NaN' when it fails, because `parseInt` is going to do that in most cases for you. (Technically it returns the global `NaN`, but when cast to a string it will be `"NaN"`.)

More to the original point, what error condition are you anticipating here, and should you really allow patron records to be created with `id` equal to "NaN"? Will that lead to a db schema type error for that column? Is that a condition in which you should fail record creation?

Reply:
@nonword after chatting with you, I think we need to send the data that we have to the app that called our service.  That means we continue responding with a 201, because that's the status we got back from Sierra, and we should include a message "No link or ID from ILS" so that we can more easily debug the stream.

```
axios.post(process.env.ILS_CREATE_PATRON_URL, req.body, {
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${ilsToken}`,
    },
  })
    .then((axiosResponse) => {
      const modeledResponse = modelResponse.patronCreator(axiosResponse.data, axiosResponse.status, req.body); // eslint-disable-line max-len
      modelStreamPatron.transformPatronRequest(req.body, modeledResponse)
        .then((streamPatronData) => {
          streamPatron(req, res, streamPatronData, modeledResponse);
        })
        .catch(() => {
          // eslint-disable-next-line max-len
          renderResponse(req, res, 201, modeledResponse); // respond with 201 even if streaming fails
        });
    })
    .catch((axiosErrorResponse) => {...
```